samples/Mvc.RestrictiveFileUpload/Controllers/UsersController.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Mvc.RestrictiveFileUpload.Models;
+
+namespace Mvc.RestrictiveFileUpload.Controllers
+{
+    public class UsersController : Controller
+    {
+        [HttpGet]
+        public IActionResult Create()
+        {
+            return View();
+        }
+
+        // Set the request form size limits *before* the antiforgery token validation filter is executed so that the
+        // limits are honored when the validation filter tries to read the form.
+        [HttpPost]
+        [RequestFormSizeLimit(bufferBodyLengthLimitFormSizeInBytes: 1024 * 1024, order: 1)]
```

Comment:
This lowers the limit from the default. We don't expect that to be the common scenario. More likely people are going to start with low limits and then raise them on an as-needed basis. You can leave this as is, but lower the default limits in Startup.ConfigureServices.